common/common-kotlin/src/main/java/com/duckduckgo/common/Domain.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2024 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.common
+
+@JvmInline
+value class Domain(
```

Comment:
Some comments:
* we have a `browser-api` module. I think this should rather go in there
* curious why `value class`?
* Do we need this wrapper just to wrap a string?

Reply:
This was already discussed on MM, but adding it here for transparency/traceability:

- _we have a browser-api module. I think this should rather go in there_
    - Done
- _Do we need this wrapper just to wrap a string?_
    - We don't strictly need the wrapper just to wrap a string. However, using the wrapper adds clarity and semantic meaning to the code. It also ensures that the type doesn't require the extra checks needed when directly dealing with a string, where additional validation is necessary due to uncertainty about its content.